cdap-app-fabric/src/main/java/co/cask/cdap/templates/AdapterSpecification.java

```diff
@@ -20,21 +20,29 @@
 import co.cask.cdap.api.data.stream.StreamSpecification;
 import co.cask.cdap.api.schedule.ScheduleSpecification;
 import co.cask.cdap.data.dataset.DatasetCreationSpec;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
 import com.clearspring.analytics.util.Preconditions;
 import com.google.common.base.Objects;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
 import com.google.gson.JsonElement;
-import com.sun.istack.Nullable;
 
+import java.util.HashSet;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * Specification of an adapter.
  */
 public final class AdapterSpecification {
+
+  private static final HashSet<ProgramType> ADAPTER_PROGRAM_TYPES = Sets.newHashSet(ProgramType.WORKFLOW,
```

Comment:
`Set<ProgramType> ALLOWED_TYPES = EnumSet.of(ProgramType.WORKER, ProgramType.WORKFLOW);`